src/nodes.coffee

```diff
@@ -1431,9 +1430,8 @@ exports.Param = class Param extends Base
     return @reference if @reference
     node = @name
     if node.this
-      node = node.properties[0].name
-      if node.value.reserved
-        node = new Literal o.scope.freeVariable node.value
+      name = "at_#{node.properties[0].name.value}"
```

Comment:
Yes, there’s a risk that it’ll get the same name as one of the helper functions such as `__slice`. If it has been used before there’s no worries: `o.scope.freeVariable` will rename the argument to `__slice1`, but if `__slice` has not been used before there will be a “reserved word __slice” error. There’s [a test for this](https://github.com/jashkenas/coffeescript/pull/3784/files#diff-e01140821ab246a31253bf5bcc14c4a3R439).